components/script/chatBar/upload/workspace.tsx

```diff
@@ -30,7 +30,9 @@ const Workspace = ({onRestart}: WorkspaceProps) => {
         setWorkspace(workspaceInput);
         setIsOpen(false);
         // Here we use selectedThreadId here because it is always set after thread is created. Thread might not be set when it is created on the fly.
-        updateThreadWorkspace(selectedThreadId ?? '', workspaceInput);
```

Comment:
This avoids the type error when workspace is changed but thread is not created. This happens when you first change workspace in the chat because thread is not created at this point.